src/link/link.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable, HttpException, HttpStatus } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { User } from '../entities/user.entity';
+import { ElderlyCode, UserInfo } from './interfaces/link.interface'
+import Hashids from 'hashids';
+
+@Injectable()
+export class LinkService {
+    constructor(
+        @InjectRepository(User)
+        private userRepository: Repository<User>,
+    ) {}
+
+    async getElderly(uid: number): Promise<UserInfo>{
+        const elderly = await this.userRepository.findOne({uid, isElderly: true});
+        if(elderly){
+            delete elderly.password;
+            return elderly;
+        }
+        throw new HttpException('Invalid uid', HttpStatus.BAD_REQUEST);
+    }
+
+    async getCaretaker(uid: number): Promise<UserInfo>{
+        const caretaker = await this.userRepository.findOne({uid, isElderly: false});
+        if(caretaker){
+            delete caretaker.password;
+            return caretaker;
+        }
+        throw new HttpException('Invalid uid', HttpStatus.BAD_REQUEST);
+    }
+
+    async getElderlyCode(uid: number): Promise<ElderlyCode> { //figure out response format
+        const user = await this.userRepository.findOne({uid});
+        console.log(user);
+        if (user) {
+            if(user['isElderly']){
+                const hashids = new Hashids('aegis', 6, 'ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890');
```

Comment:
For mine I use bcrypt to hash...maybe we should standardize it.

Reply:
We can discuss this tmr.

Reply:
Hashids is for encryption and not hashing. the encoded id of an elderly will be decoded later on. I'm not sure if bcrypt can be used for encryption or not.

Reply:
I think if Hashids function is for encryption(not hash), this is alrealy ok la